feat: add basic page for namespaces

lorenzo-cavazzi opened the pull request:
Users and groups are not first-class citizens yet. Since users may try to reach the `/projects/<namespace>` url anyway, we want to provide useful information when hitting that route.

With this PR, the existence of the `<namespace>` is tested and, according to the result (user, group or non-existing), actions are suggested to the user.

fix #750

Preview: https://lorenzotest.dev.renku.ch/

**How to test**
Try to reach a few `/projects/<namespace>` URLs, at least one for each of the 3 following types
* User: E.G. https://lorenzotest.dev.renku.ch/projects/lorenzo.cavazzi.tech
* Group: E.G. https://lorenzotest.dev.renku.ch/projects/eurobioc2019
* Non-existing: E.G. https://lorenzotest.dev.renku.ch/projects/nonexistent

rokroskar commented:
cc @emmjab @cchoirat

emmjab commented:
Thanks @lorenzo-cavazzi !

I see a typo in `group` on: https://lorenzotest.dev.renku.ch/projects/renku-qa
```
Currently, we don't provide grouü information but you can try one of the following:
```

Other than that, looks good to me!

lorenzo-cavazzi commented:
fixed, thanks! (I didn't update the deployment on `lorenzotest`)

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+  let checking = null;
+  if (props.user.fetching || props.group.fetching)
+    checking = (<div>
+      <p>We are checking if {namespace} is either a user or a group...</p>
```
Suggestion -- I think this can be a little simpler.
```<p>Searching for {namespace}...</p>```

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+    <Container fluid>
+      <Row>
+        <Col>
+          <h3>Projects for {props.namespace}</h3>
```
Suggestion: Say here if the namespace is a user or group.

```<h3>{userOrGroup} <i>{props.namespace}</i></h3>```

lorenzo-cavazzi replied:
I added `<h3>{userOrGroup} {props.namespace}</h3>` since I tried the italics for the project name but it looked weird

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+
+const NamespaceUserActions = (props) => {
+  return (<div>
+    <p>We have found a user with name {props.namespace}.</p>
```
This line is unnecessary if you take the suggestion above.

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+  return (<div>
+    <p>We have found a user with name {props.namespace}.</p>
+
+    <InfoAlert timeout={0}>
```
I don't really like the InfoAlert here. It makes the page feel more unfinished than it has to, and it makes no sense to close the InfoAlert, because then this page does not provide anything useful.

Couldn't we just do 
```
	      <ul className="mb-0">
        <li className="mb-1">
          Browse user&apos;s <Link
            className="btn btn-primary btn-sm" role="button" to={props.projectsUrl}>projects list
          </Link>
        </li>
        <li>
          Visit user&apos;s <ExternalLink url={props.gitlabUrl} size="sm" title="GitLab page" />
        </li>
      </ul>
```
And leave it at that (without the InfoAlert)?

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+
+const NamespaceGroupActions = (props) => {
+  return (<div>
+    <p>We have found a group with name {props.namespace}.</p>
```
See comments on NamespaceUserActions.

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+
+const NamespaceNotfoundActions = (props) => {
+  return (<div>
+    <p>We couldn&apos;t find any user nor group with name {props.namespace}.</p>
```
Wording:
```<p>We could not find a user or group with name {props.namespace}.</p>```

ciyer commented on src/namespace/Namespace.present.js:
```diff
@@ -0,0 +1,156 @@
+  return (<div>
+    <p>We couldn&apos;t find any user nor group with name {props.namespace}.</p>
+
+    <InfoAlert timeout={0}>
```
I think the InfoAlert makes sense in this case, so this can be left as is.

lorenzo-cavazzi commented:
@ciyer thanks for the suggestion, I implemented all of them apart from a minor difference (I commented on the change suggestion).

I am nor sure if I prefer the version with or without `<InfoAlert>`. Anyway, I updated the preview on `lorenzotest`

emmjab commented:
IMHO it doesn't change too much from a usability standpoint, still seems good to me. (I do have some usability things with https://github.com/SwissDataScienceCenter/renku-ui/issues/771, which is only tangentially related but probably will involve updating some links here too?)

vfried commented on src/api-client/instance.js:
```diff
@@ -38,6 +38,16 @@ function addInstanceMethods(client) {
+    })
+  }
 }
 
```
Shouldn't this be in user.js ?

lorenzo-cavazzi replied:
this makes definitely sense, let me move it

lorenzo-cavazzi commented:
@vfried you are right, in the future it makes sense to treat users and groups as first entity citizens and this should go under a different path. We could open another issue to keep track of that (probably an epic would me more appropriate)

ciyer commented:
I agree with both vfried and lorenzo-cavazzi . I think the `projects/{namespace}` page url should focus on projects and there should be `user/{username}` and `group/{groupname}` pages for the other purposes, but, I think it is nice to provide a link to information about the user/group since you many want that.
